aframe/utils.py

```diff
@@ -95,12 +95,33 @@ def calc_shifts_required(Tb: float, T: float, delta: float) -> int:
     return math.ceil(N)
 
 
-def get_num_shifts(
+def get_num_shifts_from_Tb(
     segments: List[Tuple[float, float]], Tb: float, shift: float
 ) -> int:
     """
     Calculates the number of required time shifts based on a list
-    of background segments
+    of background segments and the desired total background duration.
     """
     T = sum([stop - start for start, stop in segments])
     return calc_shifts_required(Tb, T, shift)
+
+
+def get_num_shifts_from_num_injections(
```

Comment:
This function also needs to account for `waveform_duration`, which gets incorporated into `buffer` and `spacing` in `calc_segment_injection_times` [here](https://github.com/ML4GW/aframev2/blob/2d928b3ec4d13849a573042fee863780fd90c24a/projects/data/data/timeslide_waveforms/utils.py#L24). Or, what might be better is re-writing that function to assume that `waveform_duration` is included in the arguments passed to it, though in that case we'd want to add some sanity checks to the given values.

Reply:
Good point, probably would rather overestimate number of timeslides we need anyway. Added that in.